view/frontend/web/internals/common.js

```diff
@@ -1,3 +1,54 @@
+var algolia = {
+	allowedHooks: [
+		'modifyAutocompleteSources',
```

Comment:
I'd use a name that describes a part of the lifecyle of your configuration like `beforeAutocomplete` or `beforeAutocompleteSource`. WDYT?